cirq/circuits/circuit_test.py

```diff
@@ -3817,3 +3817,31 @@ def test_zip():
             cirq.Circuit(cirq.X(a), cirq.CNOT(a, b)),
             cirq.Circuit(cirq.X(b), cirq.Z(b)),
         )
+
+
+def test_repr_html_escaping():
+
+    class TestGate(cirq.Gate):
+
+        def __init__(self, n_qubits, label):
+            self.n_qubits = n_qubits
+            self.label = label
+
+        def num_qubits(self):
+            return self.n_qubits
+
+        def _circuit_diagram_info_(self, args):
+            return cirq.CircuitDiagramInfo(wire_symbols=[self.label] *
+                                           self.n_qubits)
+
+    F2 = TestGate(2, "< ' F ' > ")
+    a, b = cirq.LineQubit.range(2)
+    c = cirq.NamedQubit("|c>")
+
+    circuit = cirq.Circuit([F2(a, b), F2(b, c), F2(a, c)])
+
+    # Escaping Special Characters in Gate names.
+    assert '&lt; &#x27; F &#x27; &gt;' in circuit._repr_html_()
+
+    # Escaping Special Characters in Qubit names.
+    assert '|c&gt;' in circuit._repr_html_()
```

Comment:
The file is missing a trailing newline.  (Why is this not added by the auto formatter?)

Reply:
Added one in.